_studio/shared/umc/codec/h265_dec/src/umc_h265_mfx_utils.cpp

```diff
@@ -984,6 +984,11 @@ bool CheckVideoParam_H265(mfxVideoParam *in, eMFXHWType type)
     if ((in->mfx.FrameInfo.AspectRatioW || in->mfx.FrameInfo.AspectRatioH) && !(in->mfx.FrameInfo.AspectRatioW && in->mfx.FrameInfo.AspectRatioH))
         return false;
 
+    if (in->mfx.CodecProfile == MFX_PROFILE_UNKNOWN)
+    {
+        in->mfx.CodecProfile = MatchProfile(in->mfx.FrameInfo.FourCC);
```

Comment:
This function only validates input, but doesn't change anything in provided data. @dmitry-gurulev what is the correct place for such adjustment?

Reply:
It will be great if this could be handled inside MSDK.

Reply:
@mgonchar, @fulinjie,

I've added a comment at [1576](https://github.com/Intel-Media-SDK/MediaSDK/issues/1576)